### PR TITLE
Use golang 1.6

### PR DESCRIPTION
**Pull request by dgageot**
- Use golang 1.6rc2
- Get rid of all the vendor experiment things
- Take the opportunity to clean up the dockerfile and Makefile a little

Signed-off-by: David Gageot <david@gageot.net>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,10 +1,11 @@
-FROM golang:1.5.3
+FROM golang:1.6rc2-alpine
+
+RUN apk update && apk add git
 
 RUN go get  github.com/golang/lint/golint \
             github.com/mattn/goveralls \
             golang.org/x/tools/cover \
-            github.com/tools/godep \
-            github.com/aktau/github-release
```
Comment (dmp42) [inline Dockerfile]: We don't use this anymore?

**Comment by dmp42** [comment]
LGTM

Btw, we don't need GO15VENDOREXPERIMENT anymore with 1.6 AFAIK.

**Comment by dgageot** [comment]
Not sure what's going on with go 1.6rc2 on appveyor...

**Comment by nathanleclaire** [comment]
:+1: 

We might want to just wait til the official 1.6 release to merge this -- it's coming pretty soon I think.

**Comment by dgageot** [comment]
ping @docker/core-machine-maintainers Now I used the 1.6 release

**Comment by jeanlaurent** [comment]
LGTM

**Comment by nathanleclaire** [comment]
:+1:

**Pull request merged**